### PR TITLE
docs: Add system architecture diagram + test plan outline

### PR DESCRIPTION
**Pull request by KrishanYadav333**
## Overview

Adds foundational documentation for location-proximity analysis module development.

## Files Added

- `ARCHITECTURE.md` - System architecture with Mermaid diagrams
- `docs/TEST_PLAN.md` - Comprehensive test plan (29+ test cases)
- `README.md` - Documentation formatting improvements

## Architecture Features

- Multi-dimensional proximity pipeline design
- Component integration specifications
- Performance and scalability considerations
- Technology stack documentation

## Test Strategy

- Unit tests for 4 core components
- Integration and performance tests
- Edge case validation
- Quality metrics and benchmarks

## Value

Provides technical foundation for implementation phase while maintaining professional documentation standards.

Ready for mentor review and feedback.

### CODE REVIEW

File: docs/TEST_PLAN.md [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+**Expected Output**: Outliers labeled as noise (-1)
+**Priority**: Low
+
+### Edge Cases
+
+#### Test Case: SC-EC-001
+**Description**: Insufficient data for clustering
+**Input**: Dataset with only 2 locations
+**Expected Output**: Appropriate handling (no clusters or single cluster)
+**Priority**: Medium
```
[the remark this answers is not in the document]
Comment (KrishanYadav333) [reply docs/TEST_PLAN.md]: Good catch! I've made the DBSCAN edge case more specific by clarifying the expected behavior based on the eps parameter. The test now explicitly states when points should be labeled as noise (-1) versus forming a single cluster.

File: docs/TEST_PLAN.md [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+**Description**: Identify emotional hotspot
+**Input**: Location with consistent positive sentiment (≥60% above 0.6)
+**Expected Output**: Classified as positive hotspot
+**Priority**: Medium
+
+#### Test Case: ELM-UT-004
+**Description**: Compare place types sentiment
+**Input**: Multiple churches and hospitals with sentiment data
+**Expected Output**: Statistical comparison (mean, variance, significance)
+**Priority**: Medium
```
[the remark this answers is not in the document]
Comment (KrishanYadav333) [reply docs/TEST_PLAN.md]: Excellent suggestion! I've made the statistical comparison test more concrete by specifying expected mean values and the statistical test criteria. This makes the test case much more actionable and automatable.

File: docs/TEST_PLAN.md [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+
+#### Test Case: ELM-UT-005
+**Description**: Temporal emotion trend analysis
+**Input**: Location with sentiment data over 6 months
+**Expected Output**: Trend direction (improving/declining/stable)
+**Priority**: Low
```
[the remark this answers is not in the document]
Comment (KrishanYadav333) [reply docs/TEST_PLAN.md]: Great point! I've defined quantitative criteria for the temporal trend analysis using linear regression slope thresholds. This makes the test case much more precise and automatable.

**Comment by pradeeban** [comment]
Merging, thanks!
